src/routes/accounts.py

```
@@ -1,25 +1,291 @@
+    result = await db.execute(select(PasswordResetTokenModel).where(PasswordResetTokenModel.user_id == user.id))
+    tokens = result.scalars().all()
+    token_obj = next((t for t in tokens if t.token == payload.token), None)
+
+    if not token_obj or (token_obj.expires_at.replace(tzinfo=timezone.utc) < datetime.now(timezone.utc)):
+        for token in tokens:
+            await db.delete(token)
+        await db.commit()
+        raise HTTPException(status_code=400, detail="Invalid email or token.")
+
+    try:
+        user.password = payload.password
+        for token in tokens:
+            await db.delete(token)
+        await db.commit()
+        await db.refresh(user)
+    except SQLAlchemyError:
+        await db.rollback()
+        raise HTTPException(status_code=500, detail="An error occurred while resetting the password.")
+
+    return MessageResponseSchema(message="Password reset successfully.")
+
+
+@router.post(
+    "/login/",
+    response_model=UserLoginResponseSchema,
+    status_code=status.HTTP_201_CREATED
+)
+async def login(
+    payload: UserLoginRequestSchema,
+    db: AsyncSession = Depends(get_db),
+    jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager)
+):
+    result = await db.execute(select(UserModel).where(UserModel.email == payload.email))
+    user = result.scalar_one_or_none()
+    if not user or not user.verify_password(payload.password):
+        raise HTTPException(status_code=401, detail="Invalid email or password.")
+
+    if not user.is_active:
+        raise HTTPException(status_code=403, detail="User account is not activated.")
+
+    access_token = jwt_manager.create_access_token(data={"user_id": user.id})
+    refresh_token_str = jwt_manager.create_refresh_token(data={"user_id": user.id})
+
+    refresh_token = RefreshTokenModel(user_id=user.id, token=refresh_token_str)
+    try:
+        db.add(refresh_token)
+        await db.commit()
+        await db.refresh(refresh_token)
+    except Exception:
```

You're catching a very broad `Exception` when saving the refresh token. Prefer catching `SQLAlchemyError` (or the specific DB error) so unrelated exceptions aren't swallowed. The spec mentions rolling back on DB errors and returning 500 with a specific message; catching SQLAlchemyError makes intent clearer (line with `except Exception:`).